Shared/API/CloudFunction.swift

```diff
@@ -53,22 +56,38 @@ extension CloudFunction {
 
             try Task.checkCancellation()
 
-            weakStatusables.forEach { (statusable) in
-                statusable.value?.handleEvent(status: .saved)
+            await withTaskGroup(of: Void.self) { group in
+                for weakStatusable in weakStatusables {
+                    group.addTask {
+                        guard let statusable = weakStatusable.value else { return }
+                        await statusable.handleEvent(status: .saved)
+                    }
+                }
             }
 
             // A saved status is temporary so we set it to complete after a short delay
-            delay(delayInterval) {
-                weakStatusables.forEach { (statusable) in
-                    statusable.value?.handleEvent(status: .complete)
+            Task {
+                await Task.snooze(seconds: delayInterval)
```

Comment:
Because we not in an async context, I can just tell the task to snooze instead of using our delay function.